### PR TITLE
Sidebar fixes

### PR DESCRIPTION
**Pull request by karloscastaneda**

### CODE REVIEW

File: static/sidebar/js/modal/index.js [hunk 1]
```diff
@@ -54,14 +54,19 @@
       prodNode.href = productLink;
       prodNode.innerHTML = '<div class="tvp-product-image" '+productImgStyle+'><div/>';
       thumbsFrag.appendChild(prodNode);
+
```
Comment (rfornes) [inline static/sidebar/js/modal/index.js]: Protect your code, if `title` or `price` etc is not defined code will brake
We are also using product.title few times, lets try to tackle both like:

`var trimmedTitle = productTitle.length > 50 ? productTitle.substring(0, 50 - 3) + "..." : productTitle;`

File: static/sidebar/js/modal/index.js [hunk 2]
```diff
@@ -54,14 +54,19 @@
       prodNode.href = productLink;
       prodNode.innerHTML = '<div class="tvp-product-image" '+productImgStyle+'><div/>';
       thumbsFrag.appendChild(prodNode);
+      
+      var trimmedTitle = product.title.length > 50 ? product.title.substring(0, 50 - 3) + "..." :product.title;
+      var price = product.price.toString().replace(/[^0-9.]+/g, '');
+      price = parseFloat(price).toFixed(2);
```
Comment (rfornes) [inline static/sidebar/js/modal/index.js]: What we trying to solve here in the price?

File: static/sidebar/js/modal/index.js [hunk 3]
```diff
@@ -54,14 +54,19 @@
       prodNode.href = productLink;
       prodNode.innerHTML = '<div class="tvp-product-image" '+productImgStyle+'><div/>';
       thumbsFrag.appendChild(prodNode);
+      
+      var trimmedTitle = product.title.length > 50 ? product.title.substring(0, 50 - 3) + "..." :product.title;
```
Comment (rfornes) [inline static/sidebar/js/modal/index.js]: What's the logic behind:

`substring(0, 50 - 3)`

Why do we need to substract 3? why not doing

`substring(0, 47)`

Can we handle the character qty limit as an option?

**Comment by rfornes** [comment]
@karloscastaneda same comments for what we are doing in the loader.js file

File: static/sidebar/js/modal/index.js (hunk 3 above)
Comment (karloscastaneda) [inline static/sidebar/js/modal/index.js]: handle the qty like, custom from the settings?
the "-3" was for the dots, but i guess is not necessary.

File: static/sidebar/js/modal/index.js (hunk 2 above)
Comment (karloscastaneda) [inline static/sidebar/js/modal/index.js]: first remove special characters and leave just numbres (in case there's double "$" sign)
thne if the ammount has multiple numbers ie. "59.6546" = "59.65"

File: static/sidebar/js/modal/index.js (hunk 1 above)
Comment (karloscastaneda) [inline static/sidebar/js/modal/index.js]: will do

**Comment by karloscastaneda** [comment]
@rfornes same, just to trim long titles, but since we render the modal title from there, that's why i put it in loader.js

File: static/sidebar/js/modal/index.js (hunk 2 above)
Comment (rfornes) [inline static/sidebar/js/modal/index.js]: @karloscastaneda awesome, please add a simple comment on top of that code

**Comment by karloscastaneda** [comment]
copy that @rfornes i'm about to push the changes i made

File: static/loader.js [hunk 4]
```diff
@@ -191,7 +191,8 @@
             if ('tvp_sidebar:video_click' === eventName) {
               var data = e.data;
               var selectedVideo = data.selectedVideo || {};
-              var runTime = (data.runTime || (isset(__TVPage__) ? __TVPage__ : {}) ).config[id];
+              var dataVideos = data.videos;
```
Comment (rfornes) [inline static/loader.js]: What if videos is not defined? you can do something like:

`var dataVideos = isset(data,'videos') ? data.videos : [];`

Then later on you won't need the `if (dataVideos) {` cause for loops don't brake with empty arrays.

File: static/sidebar/css/host.css [hunk 5]
```diff
@@ -99,7 +109,9 @@ div[class^="tvp-"] {
   background: whitesmoke;
   border: 0;
   -webkit-appearance: none;
-  float: right;
+  right: 3px;
+  position: absolute;
+  top: 3px;
```
Comment (rfornes) [inline static/sidebar/css/host.css]: Why was this change required?

File: static/sidebar/css/modal/styles.css [hunk 6]
```diff
@@ -128,12 +128,33 @@ body  {
   font-size: 14px;
   text-decoration: none;
   margin: 10px 0;
+  color: #1F1D26;
+}
+
+.tvp-product-title .tvp-tooltip {
```
Comment (rfornes) [inline static/sidebar/css/modal/styles.css]: tooltip will be a custom thing for Bleeping, please just remove it from this branch so it doesn't stop your fixes to go live

File: static/loader.js (hunk 4 above)
Comment (karloscastaneda) [inline static/loader.js]: copy!

File: static/sidebar/css/host.css (hunk 5 above)
Comment (karloscastaneda) [inline static/sidebar/css/host.css]: this was for moving the "close button" to the top/right corner of the modal

File: static/sidebar/css/modal/styles.css (hunk 6 above)
Comment (karloscastaneda) [inline static/sidebar/css/modal/styles.css]: copy!
